host/xtest/adbg/src/adbg_run.c

```diff
@@ -100,7 +100,7 @@ int Do_ADBG_AppendToSuite(
 		snprintf(p, size, "%s+%s", Dest_p->SuiteID_p,
 			 Source_p->SuiteID_p);
 	else
-		strncpy(p, Source_p->SuiteID_p, size);
+		snprintf(p, size, "%s", Source_p->SuiteID_p);
```

Comment:
This seems to be fixing a false positive. The compiler doesn't seem to understand that `p` is also allocated based on `size.

Reply:
I agree. If it was suspicious, `snprintf()` should not match neither.
Interesting article on that from [readhat](https://developers.redhat.com/blog/2018/05/24/detecting-string-truncation-with-gcc-8/):

> _Using strncpy Safely_
In general, it is not possible to avoid string truncation by strncpy except by sizing the destination to be at least a byte larger than the length of the source string. With that approach, however, **using strncpy becomes unnecessary and the function can be avoided in favor of other APIs such as strcpy or (less preferably) memcpy.** Much has been written about the problems with strncpy and we recommend to avoid it whenever possible. It is, however, worth keeping in mind that unlike other standard string-handling functions, strncpy always writes exactly as many characters as specified by the third argument; if the source string is shorter, the function fills the remaining bytes with NULs.

From the bold part, we should use `strcpy` !!! :) the one prohibited!
I think `snprintf` is fine and consistent with alternate case. Let's use it.

Reply:
As far as I can tell was the code already correct so we're not really fixing anything. I think the commit message should be updated to reflect that.

Reply:
ok, will do.